plugin-build/build.gradle.kts

```diff
@@ -70,3 +70,41 @@ if (gradle.gradleVersion >= "6.6.0") {
     val publish = extensions.getByType(MavenPublishPluginExtension::class.java)
     publish.releaseSigningEnabled = BuildUtils.shouldSignArtifacts()
 }
+
+apply<DistributionPlugin>()
+
+val sep = File.separator
+
+configure<DistributionContainer> {
+    this.getByName("main").contents {
+        from("build${sep}libs")
+        from("build${sep}publications${sep}maven")
+        from("build${sep}publications${sep}sentryPluginPluginMarkerMaven")
+        duplicatesStrategy = DuplicatesStrategy.INCLUDE
+    }
+}
+
+tasks.named("distZip").configure {
+    this.dependsOn("publishToMavenLocal")
+    this.doLast {
+        val distributionFilePath = "${this.project.buildDir}${sep}distributions" +
+            "${sep}${this.project.name}-${this.project.version}.zip"
+        val file = File(distributionFilePath)
+        if (!file.exists()) {
+            throw IllegalStateException(
+                "Distribution file: $distributionFilePath does not exist"
+            )
+        }
+        if (file.length() == 0L) {
+            throw IllegalStateException("Distribution file: $distributionFilePath is empty")
+        }
+    }
+}
+
+afterEvaluate {
+    configure<MavenPublishPluginExtension> {
+        // signing is done when uploading files to MC
+        // via gpg:sign-and-deploy-file (release.kts)
+        releaseSigningEnabled = false
+    }
+}
```

Comment:
I believe this is unnecessary. Specifically the task should not depend on `publishToMavenLocal` (alternatively on `assemble`).
You should also not check the size of the output. If something goes wrong Gradle will fail :)

Reply:
Depending on `publishToMavenLocal` is required. If not, the first build won't generate `distributions/` and you'll need to run it again.

Reply:
the build should fail if it's not generated as our release process depends on it.

Reply:
I understand. Not elegant but yeah ¯\_(ツ)_/¯

Also a couple of things that are still worth fixing:
1. This whole code should go inside the if-then-else where the maven.publish plugin is applied (check for Gradle 6.x). This because `distribution` now depends on `maven.publish` and it should be applied only on Gradle 6.x
2. I still believe the whole `this.doLast` block can be removed. If you want to be sure that the task is actually executing you can replace it with a:
```kotlin
if(inputs.sourceFiles.isEmpty) throw StopExecutionException("...")
``` 

That will prevent the `distZip` fail to return with a `NO-SOURCE` result if the `publishToMavenLocal` produces no output.

Reply:
The task is being run, the issue is the timing. If the `doLast` block is removed in the latest changes (`fef85c5`), it checks whether the file exists before creating it. I don't know how the task execution is scheduled, nor how they are optimized (if they are); but that task seems to be required to execute last (to not run before the `zip` is created). Let me know if you dive more into that.

Reply:
Yup I see what's the problem here. I've pushed a branch with the fix: https://github.com/cortinico/sentry-android-gradle-plugin/tree/nc/build-artifacts

Specifically on https://github.com/cortinico/sentry-android-gradle-plugin/commit/8d11c3b4aca92f33b9fea7ead9b2dbce1069bd9d

You can use `onlyIf` that gets evaluated at execution time to do the job.
I've verified it locally with:

```
$ rm -rf plugin-build/build && ./gradlew -p plugin-build distZip && ls plugin-build/build/distributions

To honour the JVM settings for this build a single-use Daemon process will be forked. See https://docs.gradle.org/7.0.2/userguide/gradle_daemon.html#sec:disabling_the_daemon.
Daemon will be stopped at the end of the build

> Task :dokkaHtml
Initializing plugins
Dokka is performing: documentation for sentry-android-gradle-plugin
Validity check
Creating documentation models
Transforming documentation model before merging
Merging documentation models
Transforming documentation model after merging
Creating pages
Transforming pages
Rendering

> Task :compileKotlin
w: Runtime JAR files in the classpath should have the same version. These files were found in the classpath:
    [...]
w: Consider providing an explicit dependency on kotlin-reflect 1.4 to prevent strange errors
w: Some runtime JAR files in the classpath have an incompatible version. Consider removing them from the classpath

> Task :publishPluginMavenPublicationToMavenLocal
Multiple publications with coordinates 'io.sentry:sentry-android-gradle-plugin:2.1.1-SNAPSHOT' are published to repository 'mavenLocal'. The publications will overwrite each other!

BUILD SUCCESSFUL in 28s
17 actionable tasks: 17 executed
sentry-android-gradle-plugin-2.1.1-SNAPSHOT.zip
```

Reply:
Tried to play around with it for a while to see if I could improve it a bit, but didn't manage, so using your approach 👍.